services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/seven_zip_wrapper.py

```diff
@@ -0,0 +1,61 @@
+import asyncio
+import logging
+import re
+from pathlib import Path
+
+from models_library.basic_types import IDStr
+from servicelib.progress_bar import ProgressBarData
+
+from ..core.errors import SevenZipError
+from ..core.utils import async_command
+
+_logger = logging.getLogger(__name__)
+
+
+async def _get_file_count(zip_path: Path) -> int:
+    result = await async_command(f"7z l {zip_path}")
+    if not result.success:
+        raise SevenZipError(command=result.command, command_result=result.message)
+
+    match = re.search(r"\s*(\d+)\s*files", result.message)
+    return int(match.group().replace("files", "").strip() if match else "0")
+
+
+async def unarchive_zip_to(
+    zip_path: Path,
+    output_dir: Path,
+    progress_bar: ProgressBarData | None = None,
+) -> set[Path]:
+    if not progress_bar:
+        progress_bar = ProgressBarData(
+            num_steps=1, description=IDStr(f"extracting {zip_path.name}")
+        )
+
+    file_count = await _get_file_count(zip_path)
+
+    command = f"7z x {zip_path} -o{output_dir} -bb1"
+    process = await asyncio.create_subprocess_shell(
+        command, stdout=asyncio.subprocess.PIPE, stderr=asyncio.subprocess.PIPE
+    )
+    assert process.stdout  # nosec
+    assert process.stderr  # nosec
+
+    async with progress_bar.sub_progress(
+        steps=file_count, description=IDStr("...")
```

Comment:
Not the first time I see using `IDStr` for descriptions ...  
NOTE that:
1.  Calling `IDStr("...")` is equivalent to calling `str("..."). Therefore in this context: it basically does nothing. To perform a validation you need to create a `TypeAdapter(IDStgr).validate_python("...")` etc 
2. If you want to annotate a field . Then note that `IDStr` is intended for annotation of **St**ring **ID**entifiers (e.g. a `prj_123456`) and not descriptions! There are other constraints for e.g.  `ShortTruncatedStr` or `LongTruncatedStr`

Reply:
@pcrespov I think this is a flaw in the ProgressBar class. I guess I did it wrong at the time. but indeed I would just remove that IDStr from there it makes no sense. But probably should go in a separate PR, maybe an issue for this would be nice.